modules/imap/hm-ews.php

```diff
@@ -522,58 +547,191 @@ public function get_message_list($itemIds, $include_preview = false) {
+            }
+        }
+        $msg['charset'] = $cset;
+        $msg['preview_msg'] = $include_preview ? strip_tags($message->getBody()) : "";
+
+        return $msg;
+    }
+
+    /**
+     * Detect the type of EWS message/item
+     */
+    protected function detectMessageType($message) {
+        if ($message instanceof Type\CalendarItemType) {
+            return 'calendar';
+        } elseif ($message instanceof Type\MeetingRequestMessageType ||
+                  $message instanceof Type\MeetingResponseMessageType ||
+                  $message instanceof Type\MeetingCancellationMessageType) {
+            return 'meeting_request';
+        } elseif ($message instanceof Type\MessageType) {
+            return 'message';
+        } else {
+            return 'unknown';
+        }
+    }
+
+    /**
+     * Extract properties specific to regular messages
+     */
+    protected function getRegularMessageProperties($message) {
+        return [
+            'internal_date' => $message->getDateTimeCreated() ?
+                $message->getDateTimeCreated()->format('Y-m-d H:i:s.u') : date('Y-m-d H:i:s.u'),
+            'size' => $message->getSize() ?: 0,
+            'date' => $message->getDateTimeReceived() ?
+                $message->getDateTimeReceived()->format('Y-m-d H:i:s.u') : date('Y-m-d H:i:s.u'),
+            'from' => $message->getFrom() ? $this->extract_mailbox($message->getFrom()) : '',
+            'to' => $message->getToRecipients() ? $this->extract_mailbox($message->getToRecipients()) : '',
+            'subject' => $message->getSubject() ?: '',
+            'content-type' => null,
+            'references' => $message->getReferences() ?: null,
+            'message_id' => $message->getInternetMessageId() ?: null,
+        ];
+    }
+
+    /**
+     * Extract properties specific to calendar items
+     */
+    protected function getCalendarProperties($message) {
+        $props = [
+            'internal_date' => $message->getDateTimeCreated() ?
+                $message->getDateTimeCreated()->format('Y-m-d H:i:s.u') : date('Y-m-d H:i:s.u'),
+            'size' => $message->getSize() ?: 0,
+            'date' => $message->getDateTimeReceived() ?
+                $message->getDateTimeReceived()->format('Y-m-d H:i:s.u') :
+                ($message->getStart() ? $message->getStart()->format('Y-m-d H:i:s.u') : date('Y-m-d H:i:s.u')),
+            'from' => $message->getOrganizer() ? $this->extract_mailbox($message->getOrganizer()) : '',
+            'to' => $message->getRequiredAttendees() ? $this->extract_mailbox($message->getRequiredAttendees()) : '',
+            'subject' => $message->getSubject() ?: '',
+            'content-type' => null,
+            'references' => null,
+            'message_id' => null,
+            // Calendar-specific properties
+            'calendar_start' => $message->getStart() ? $message->getStart()->format('Y-m-d H:i:s') : null,
```

Comment:
Are these additional properties going to be used somewhere? If they are not, I prefer we don't return extra data, use more memory and make it slower just for the sake of returning what is provided by EWS.